superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -326,7 +327,19 @@ class AnnotationLayer extends React.PureComponent {
                     metadata && metadata.canBeAnnotationType(annotationType)
                   );
                 })
-                .map(x => ({ value: x.id, label: x.title, slice: x })),
+                .map(x => ({
+                  value: x.id,
+                  label: x.title,
+                  slice: {
+                    ...x,
+                    data: {
+                      ...x.data,
+                      groupby: x.data.groupby.map(metricOrColumn =>
+                        getMetricLabel(metricOrColumn),
+                      ),
```

Comment:
I know this is can be a pain to grasp from the JS codebase, but the type of `groupby` is actually `QueryFormColumn`, and the correct label getter is `getColumnLabel`:

```suggestion
                      groupby: x.data.groupby.map(column =>
                        getColumnLabel(column),
                      ),
```